### ISSUE TITLE
Is it possible to set the timezone of the container?

### ISSUE THREAD

**Issue by oliroe** [fmartinou/whats-up-docker#58]
I have WUD set to run on the following schedule but noticed it ran at 7am this morning
0 */6 * * *

Is it possible to set the timezone of the container? Linuxserver packages allow an environment variable of TZ=Europe/London as an example.

Thanks

**Comment by fmartinou** [comment]
Hi,
The container is UTC by default but you should be able to change the TZ by mounting `/etc/localtime` from your host machine.

#### Example:
```bash
# My host is CEST
$ date
ven. 06 août 2021 09:35:01 CEST

# If I run the container without any options, date is UTC
$ docker container run --rm fmartinou/whats-up-docker:5.2.0 ash -c date
Fri Aug  6 07:35:12 UTC 2021

# If I mount /etc/localtime, then I get the same TZ as my host
$ docker container run --rm -v /etc/localtime:/etc/localtime:ro fmartinou/whats-up-docker:5.2.0 ash -c date
Fri Aug  6 09:35:26 CEST 2021
```

**Comment by oliroe** [comment]
Thanks for the pointer, will check it out

I suppose in an ideal world it would be possible to set the timezone as anything rather than just the host/UTC

**Comment by fmartinou** [comment]
> I suppose in an ideal world it would be possible to set the timezone as anything rather than just the host/UTC

IMHO, this is not very common; either the server is configured with local time or with UTC.
I'd consider other situations more like bad practices than valid use-cases; don't you?

**Comment by oliroe** [comment]
The only reason I can think of is if you use a VPS. That server could be hosted anywhere and the time zone of host/UTC wouldn't help. Not a major issue as you say

**Comment by thomazz-nl** [comment]
Timezones can be tricky to deal with, and it took me a bit to figure it out what makes Alpine harder to deal with. It is because they (by default) miss the TZDATA package. So the options are:
1. Install TZDATA package in the image/container.
2. Mount /etc/localtime (and perhaps /etc/timezone) from host.
3. Supply the TZ data in POXIX notation using the environment variable TZ. Add to compose file:
```
        environment:
            TZ: CET-1CEST,M3.5.0,M10.5.0/3
```
To know what you have to fill in here, check the contents of the file for your timezone on a system that does have TZDATA present.

**Comment by fmartinou** [comment]
Thanks for your analysis @thomazz-nl !

I added the package `tzdata` to the image (the image is now 1MB bigger gzipped so not a big deal).
Now you can set the timezone using the regular `TZ` env var.

I also changed the logger so the `time` in log lines also reflects the local time (which remains UTC by default).

The [draft of the documentation is here](https://github.com/fmartinou/whats-up-docker/tree/develop/docs/configuration/timezone).

For the moment, the feature is available in `fmartinou/whats-up-docker:feature__58`
Let me know if everything is ok on your end.

**Comment by oliroe** [comment]
Just tested adding the TZ variable and everything works as expected. Good work all round.

Thanks

**Comment by fmartinou** [comment]
Great!

It's available in version `develop` and will be part of the next release (probably a `5.3.0`).

**Comment by fmartinou** [comment]
Available in `5.3.0`.
